tools/codegen/dest/lazy_ir.py

```diff
@@ -355,3 +358,75 @@ def __call__(self, f: NativeFunction) -> List[str]:
             return ["\n".join([f"{shape_sig.shape_decl};"])]
         else:
             return []
+
+
+@dataclass
+class LazyIrBuilder:
+    schema: NonNativeLazyIrSchema
```

Comment:
I guess i'm wondering (generally) why you needed a new schema type.  i will probably start to realize why when i read more of the code..

Reply:
same reason as above. It would just be more difficult to construct a `LazyIrSchema` given what I had. Which is why I did the above.

However, I might be able to make `NonNativeLazyIrSchema` a subclass of `LazyIrSchema` just to make things a bit nicer.